### PR TITLE
Add AWS release v11.4.0 for HA masters

### PR DESCRIPTION
**Pull request by marians**
Main purpose of this release is to make [high availability of master nodes ("HA masters")](https://github.com/giantswarm/giantswarm/issues/8346) generally available on AWS.

## TODO

- [x] Link to the final aws-/cluster-operator versions used from release notes, set version numbers in release.yaml
- [x] Set status to `active`, adjust the release date

## Component changes compared to 11.3.x

- aws-operator
- chart-operator
- cluster-operator
- cluster-autoscaler
- calico-node
- cert-exporter
- cert-manager
- etcd
- external-dns
- net-exporter

### CODE REVIEW

File: aws/v11.4.0/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+## :zap: Giant Swarm Release v11.4.0 for AWS :zap:
+
+This release introduces master node high availability, which means that clusters will have three master nodes by default in three different availability zones (or ).
+
+- Multiple master nodes are only supported in AWS regions having at least three availability zones. This excludes China/Beijing (`cn-north-1`).
+
+**Note for SEs:** This release contains an external-dns fix introduced in [11.3.2](https://github.com/giantswarm/releases/blob/master/aws/v11.3.2/release-notes.md). It requires manual intervention for cluster upgrades in China to work. When upgrading a cluster, existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred. After upgrading, delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets.
+
+**Note for future 11.3.x releases:** Until upstream external-dns issue is fixed, please include this note and the one above in all future 11.3.x releases.
```
Comment (marians) [inline aws/v11.4.0/release-notes.md:9]: @cokiengchiara Are these notices actually required in 11.4.x?

Comment (cokiengchiara) [reply aws/v11.4.0/release-notes.md:9]: I think yes, for China cluster upgrades (but not creation) from 11.3.x to 11.4.0. Can @sslavic confirm?

File: aws/v11.4.0/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+## :zap: Giant Swarm Release v11.4.0 for AWS :zap:
+
+This release introduces master node high availability, which means that clusters will have three master nodes by default in three different availability zones (or ).
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:3]:
```suggestion
This release introduces master node high availability, which means that clusters will have three master nodes by default in three different availability zones.
```

File: aws/v11.4.0/release-notes.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+- Multiple master nodes are only supported in AWS regions having at least three availability zones. This excludes China/Beijing (`cn-north-1`).
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:5]: This is not true anymore.

File: aws/v11.4.0/release-notes.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+**Note for SEs:** This release contains an external-dns fix introduced in [11.3.2](https://github.com/giantswarm/releases/blob/master/aws/v11.3.2/release-notes.md). It requires manual intervention for cluster upgrades in China to work. When upgrading a cluster, existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred. After upgrading, delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets.
+
+**Note for future 11.3.x releases:** Until upstream external-dns issue is fixed, please include this note and the one above in all future 11.3.x releases.
+
+## Change details
+
+### aws-operator vX.X.X
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:13]:
```suggestion
### aws-operator v8.7.0
```

File: aws/v11.4.0/release-notes.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+- Several changes to support running clusters with multiple master nodes as well as migrating from a single to multiple masters.
+- Several improvements regarding the deletion of  tenant clusters and related AWS resources.
+- Kubernetes master nodes in tenant clusters will now receive the label `giantswarm.io/control-plane` in metadata.
+- Fixed Prometheus metrics gathering from calico endpoint.
+- The Docker image now uses alpine v3.12 as a base.
+
+### cluster-operator vX.X.X
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:21]:
```suggestion
### cluster-operator v2.3.0
```

File: aws/v11.4.0/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+**Note for SEs:** This release contains an external-dns fix introduced in [11.3.2](https://github.com/giantswarm/releases/blob/master/aws/v11.3.2/release-notes.md). It requires manual intervention for cluster upgrades in China to work. When upgrading a cluster, existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred. After upgrading, delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets.
+
+**Note for future 11.3.x releases:** Until upstream external-dns issue is fixed, please include this note and the one above in all future 11.3.x releases.
+
+## Change details
+
+### aws-operator vX.X.X
+
+- Several changes to support running clusters with multiple master nodes as well as migrating from a single to multiple masters.
+- Several improvements regarding the deletion of  tenant clusters and related AWS resources.
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:16]:
```suggestion
- Several improvements regarding the deletion of tenant clusters and related AWS resources.
```

File: aws/v11.4.0/release-notes.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+- Kubernetes master nodes in tenant clusters will now receive the label `giantswarm.io/control-plane` in metadata.
+- Fixed Prometheus metrics gathering from calico endpoint.
+- The Docker image now uses alpine v3.12 as a base.
+
+### cluster-operator vX.X.X
+
+- Several improvements regarding the deletion of  tenant clusters and related AWS resources.
```
Comment (xh3b4sd) [inline aws/v11.4.0/release-notes.md:23]:
```suggestion
- Several improvements regarding the deletion of tenant clusters and related AWS resources.
```

File: aws/v11.4.0/release.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  - name: node-exporter
+    componentVersion: 0.18.1
+    version: 1.2.0
+  components:
+  - name: app-operator
+    version: 1.0.0
+  - name: aws-cni
+    version: 1.6.0
+  - name: aws-operator
+    version: 8.6.2-dev
```
Comment (xh3b4sd) [inline aws/v11.4.0/release.yaml:47]:
```suggestion
    version: 8.7.0
```

File: aws/v11.4.0/release.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  - name: cert-operator
+    version: 0.1.0
+  - name: cluster-operator
+    version: 2.2.1-dev
```
Comment (xh3b4sd) [inline aws/v11.4.0/release.yaml:51]:
```suggestion
    version: 2.3.0
```

File: aws/v11.4.0/release-notes.md (hunk 3 above)
Comment (marians) [reply aws/v11.4.0/release-notes.md:5]: Changed

File: aws/v11.4.0/release-notes.md (hunk 1 above)
Comment (marians) [reply aws/v11.4.0/release-notes.md:7]: @sslavic Would be nice if you could confirm this, as we are approaching the release

@cokiengchiara I guess the `Note for future 11.3.x releases` part needs adaptation to 11.4.x, right?

Comment (cokiengchiara) [reply aws/v11.4.0/release-notes.md:7]: > @cokiengchiara I guess the `Note for future 11.3.x releases` part needs adaptation to 11.4.x, right?

If I understand correctly, yes. But would love @sslavic to confirm. Just pinged him in Slack about it too.

Comment (sslavic) [reply aws/v11.4.0/release-notes.md:7]: @marians @cokiengchiara 

Note is applicable if:
- AWS platform release 11.4.0 makes use of external-dns-app v1.2.1, and
- Customer can upgrade an existing cluster from older node pools 10+ platform release with older external-dns-app, to 11.4.0, and if
- Cluster being upgraded is in AWS China (currently limited to few Adidas clusters).

external-dns-app v1.2.1 introduced a fix to a problem Adidas was experiencing, external-dns configuration was changed to prefer CNAME. It works automatically (creates route53 records) for the new clusters, but for cluster upgrades CNAME preference is not respected (existing route53 records do not get automatically updated, have to be manually dropped first, and then external-dns creates new valid ones) - therefore the note with instructions how to get if fixed, and also meta-note to transfer the note to a future release notes too.

Issue has been reported upstream, https://github.com/kubernetes-sigs/external-dns/issues/1617
It's not getting any traction from the community. Likely it's hurting us the most. We should consider investing time and effort to contribute the fix.

Maybe another alternative is to ensure together with @jgsqware that no more clusters get created with AWS [10 - 11.3.1], so without CNAME preference. Then we need also to ensure all the existing clusters in AWS China are upgraded to a newer release and have the manual workaround applied. If both of that happens, proactively or eventually, I guess we can drop the note about manual upgrade step even before/without upstream bug fix.

Comment (cokiengchiara) [reply aws/v11.4.0/release-notes.md:7]: thanks, then the answer for if this note should persist is yes. @sslavic regarding us working on external dns, provided my thoughts here https://github.com/giantswarm/giantswarm/issues/11606

Comment (marians) [reply aws/v11.4.0/release-notes.md]: Thanks!

**Pull request merged**